src/components/NumericStepper/__tests__/NumericStepper.spec.tsx

```diff
@@ -0,0 +1,278 @@
+        const stepperButtons = screen.getAllByRole('button');
+
+        expect(inputButton).toHaveAttribute('min', '2');
+        expect(inputButton).toHaveAttribute('max', '6');
+        expect(inputButton).toHaveAttribute('step', '2');
+        expect(inputButton).toHaveAttribute('style', 'width: 4ch;');
+        expect(inputButton).toHaveAttribute('value', '4');
+        expect(stepperButtons[0]).not.toHaveAttribute('disabled');
+        expect(stepperButtons[1]).not.toHaveAttribute('disabled');
+    });
+
+    it('should react on stepDown click', async () => {
+        view = render(<NumericStepper onChange={onChangeMock} step={2} defaultValue={4} />);
+        const inputButton = screen.getByRole('spinbutton');
+
+        // Decrease by 2 (to equal to 2) and make sure that number of onChange calls equal to 1
+        await decreaseClick();
+
+        expect(onChangeMock).toBeCalledTimes(1);
+        expect(onChangeMock).toBeCalledWith(2);
+        expect(inputButton).toHaveAttribute('value', '2');
+        expect(onChangeMock).toHaveBeenCalledWith(2);
+
+        // Decrease by 2 (to equal to 0) and make sure that number of onChange calls equal to 2
+        await decreaseClick();
+
+        expect(onChangeMock).toBeCalledTimes(2);
+        expect(onChangeMock).toBeCalledWith(0);
+        expect(inputButton).toHaveAttribute('value', '0');
+        expect(onChangeMock).toHaveBeenCalledWith(0);
+    });
+
+    it('should react on stepUp click', async () => {
+        view = render(<NumericStepper onChange={onChangeMock} step={2} />);
+        const inputButton = screen.getByRole('spinbutton');
+
+        // Increase by 2 (to equal to 2) and make sure that number of onChange calls equal to 1
+        await increaseClick();
+
+        expect(onChangeMock).toBeCalledTimes(1);
+        expect(onChangeMock).toBeCalledWith(2);
+        expect(inputButton).toHaveAttribute('value', '2');
+        expect(onChangeMock).toHaveBeenCalledWith(2);
+
+        // Increase by 2 (to equal to 4) and make sure that number of onChange calls equal to 2
+        await increaseClick();
+
+        expect(onChangeMock).toBeCalledTimes(2);
+        expect(onChangeMock).toBeCalledWith(4);
+        expect(inputButton).toHaveAttribute('value', '4');
+        expect(onChangeMock).toHaveBeenCalledWith(4);
+    });
+
+    it('should disable buttons when limits are reached', async () => {
+        const user = userEvent.setup();
+        view = render(
+            <NumericStepper onChange={onChangeMock} minValue={1} maxValue={3} defaultValue={2} />
+        );
+        const inputButton = screen.getByRole('spinbutton');
+        const stepperButtons = screen.getAllByRole('button');
+
+        // Simulate two clicks on stepUp button
+        await increaseClick();
+
+        expect(inputButton).toHaveAttribute('value', '3');
+        expect(onChangeMock).toBeCalledTimes(1);
+        expect(stepperButtons[1]).toHaveAttribute('disabled');
```

Comment:
maybe check that the other button isn't disabled?

Reply:
:+1: